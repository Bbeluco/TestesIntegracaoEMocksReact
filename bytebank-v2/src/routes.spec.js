import { render, screen } from '@testing-library/react';
import { BrowserRouter, MemoryRouter, Routes, Route } from 'react-router-dom';
import App from './paginas/Principal/App';
import Cartoes from './componentes/Cartoes/index';
import AppRoutes from './routes';

describe('Rotas', () => {
  test('Rota principal', () => {
    render(<App />, { wrapper: BrowserRouter });
    const rotaPrincipal = screen.getByText('Olá, Joana :)!');
    expect(rotaPrincipal).toBeInTheDocument();
  });

  test('Rota de cartoes', () => {
    const rota = '/cartoes';

    render(
      <MemoryRouter initialEntries={[rota]}>
        <Routes>
          <Route path="/" element={<App />}>
            <Route path="cartoes" element={<Cartoes />} />
          </Route>
        </Routes>
      </MemoryRouter>
    );

    const tituloPaginaCartoes = screen.getByTestId('titulo-principal');
    expect(tituloPaginaCartoes).toHaveTextContent('Meus cartões');
  });

  test('Deve renderizar localizacao da rota atual', () => {
    const rota = '/cartoes';

    render(
      <MemoryRouter initialEntries={[rota]}>
        <App />
      </MemoryRouter>
    );
    const localizacao = screen.getByTestId('location');
    expect(localizacao).toHaveTextContent(rota);
  });

  test('Deve renderizar pagina de 404', () => {
    const rota = '/extrato';

    render(
      <MemoryRouter initialEntries={[rota]}>
        <AppRoutes />
      </MemoryRouter>
    );

    const pagina404 = screen.getByTestId('pagina-404');
    expect(pagina404).toMatchSnapshot();
  });
});
