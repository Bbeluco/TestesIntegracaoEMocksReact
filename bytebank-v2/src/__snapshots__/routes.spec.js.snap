// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rotas Deve renderizar pagina de 404 1`] = `
<div
  class="container"
  data-testid="pagina-404"
>
  <h1>
    Ops! Não encontramos a página
  </h1>
  <p>
    E olha que exploramos o universo procurando por ela! Que tal voltar e tentar novamente?
  </p>
  <a
    class="link"
    href="/"
  >
    Voltar ao início
  </a>
  <img
    alt="Imagem ilustrativa do espaço"
    src="404.svg"
  />
</div>
`;
